controllers/progressiverollout_controller_test.go

```diff
@@ -219,14 +219,44 @@ var _ = Describe("ProgressiveRollout Controller", func() {
 
 	Describe("Reconciliation loop", func() {
 		It("should reconcile", func() {
-			By("creating a progressive rollout object")
+			By("creating a cluster")
+			cluster := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: "single-stage-cluster", Namespace: namespace, Labels: map[string]string{utils.ArgoCDSecretTypeLabel: utils.ArgoCDSecretTypeCluster}},
+				Data: map[string][]byte{
+					"server": []byte("https://single-stage-pr.kubernetes.io"),
+				},
+			}
+			Expect(k8sClient.Create(ctx, cluster)).To(Succeed())
```

Comment:
This test was to test the full reconciliation loop.

Here we create:
- an ArgoCD application
- an ArgoCD cluster - a k8s secret

Note that we don't really need the actual ApplicationSet because we can manually add the owner reference in the Application.

Once we have 1 App and 1 cluster, we create a ProgressiveRollout that should sync that Application. At the end of the reconciliation loop we expect the ProgressiveRollout object status is `succeded` as per https://github.com/Skyscanner/argocd-progressive-rollout/blob/2c186ce9acbeb1b69314c97bf2c650fb99cb04d8/controllers/progressiverollout_controller_test.go#L243

Once we add more functionalities into the controller, we might want to change this test.

Does it make sense?